From 28af64664e6d76b80449fcf9e24250e78907d48c Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Thu, 2 Jun 2016 23:36:32 -0500
Subject: [PATCH 113/131] fix initcall timestamps

Print more finegrained initcall timings

use the tsc instead of the jiffies clock for initcall_debug
----
 init/main.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)
---
 init/main.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/init/main.c b/init/main.c
index 997908e6d8a3..86d94caa1526 100644
--- a/init/main.c
+++ b/init/main.c
@@ -830,27 +830,27 @@ __setup("initcall_blacklist=", initcall_blacklist);
 static __init_or_module void
 trace_initcall_start_cb(void *data, initcall_t fn)
 {
-	ktime_t *calltime = (ktime_t *)data;
+	unsigned long long *calltime = (unsigned long long *)data;
 
-	printk(KERN_DEBUG "calling  %pF @ %i\n", fn, raw_smp_processor_id());
-	*calltime = ktime_get();
+	printk(KERN_DEBUG "calling  %pF @ %i\n", fn, task_pid_nr(current));
+	*calltime = local_clock();
 }
 
 static __init_or_module void
 trace_initcall_finish_cb(void *data, initcall_t fn, int ret)
 {
-	ktime_t *calltime = (ktime_t *)data;
-	ktime_t delta, rettime;
+	unsigned long long *calltime = (unsigned long long *)data;
+	unsigned long long delta, rettime;
 	unsigned long long duration;
 
-	rettime = ktime_get();
-	delta = ktime_sub(rettime, *calltime);
-	duration = (unsigned long long) ktime_to_ns(delta) >> 10;
+	rettime = local_clock();
+	delta = rettime - *calltime;
+	duration = delta >> 10;
 	printk(KERN_DEBUG "initcall %pF returned %d after %lld usecs\n",
 		 fn, ret, duration);
 }
 
-static ktime_t initcall_calltime;
+static unsigned long long initcall_calltime;
 
 #ifdef TRACEPOINTS_ENABLED
 static void __init initcall_debug_enable(void)
-- 
2.20.1

